versions/protocol-2-x.md

```diff
@@ -1465,6 +1466,123 @@ _Response_
 * result: `SymbolInformation[]` as defined above.
 * error: code and message set in case an exception happens during the workspace symbol request.
 
+#### <a name="workspace_documentation"></a>Workspace Documentation Request
+
+The workspace documentation request is sent from the client to the server to list project-wide documentation.
+
+_Request_
+* method: 'workspace/documentation'
+* params: `WorkspaceDocumentationParams` defined as follows:
+```typescript
+/**
+ * The parameters of a Workspace Documentation Request.
+ */
+interface WorkspaceDocumentationParams {
+	/**
+	 * An optional non-empty query string, to be matched against the `name` of
+	 * `Documentation` in the workspace. For example, a user may query `HTTP Request`
+	 * and the server may respond with the `Documentation` named `Request`
+	 * which is itself a child of `Documentation` named `HTTP`.
+	 */
+	query?: string;

```

Comment:
Am I missing something or does this reference a field `name` that doesnt exist on `Documentation`? :sweat_smile: 

I have slight concerns over how this is meant to be structured. It feels like what its trying to convey is that implementations can parse it as they see fit and perform some kind of (fuzzy? not fuzzy? both?) matching on its interpretation of how it was parsed.

Reply:
Ah, `name` should be `label` in the docstring here.

w.r.t. `query` itself: you may be shocked to learn that `workspace/symbol` has a `query` field that is.. completely undefined, then. https://github.com/slimsag/language-server-protocol/blob/18e90e8ba5d5349d56233867b7ccfebbc868a6ea/versions/protocol-2-x.md#workspace-symbols-request

My goal here was to hint that there is an implicit nested structure of `Documentation`, and language servers would do well to utilize that structure in implementing their query syntax - but still, just like `workspace/symbol` - it's totally arbitrary and up to the server to decide.

In practice, this may not matter if we go with the LSIF implementation (because then querying is not a part of the spec at all, but rather something we decide on the Sourcegraph side.) Ultimately, my goal is to just enable us to have something like Olaf's "blazingly fast symbol search" proposal in the future: https://docs.google.com/document/d/1p3Bm21a4SQicbZ-ojVTVYN6FupwNk8HuJPoSGDSmOm0/edit

Reply:
>  it's totally arbitrary and up to the server to decide.

To make matters worse, VS Code takes the liberty to filter out `workspace/symbol` results that don't match the undocumented/unspecified rules of VS Code. VS Code even frequently changes this filtering logic. I implemented a ton of useful query features in the Scala language server that no longer work in VS Code (but work in other editors).